plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/RewriteCaseSensitiveComparison.java

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.expression;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.base.expression.ConnectorExpressionRule;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.QueryParameter;
+import io.trino.spi.expression.Call;
+import io.trino.spi.expression.FunctionName;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.VarcharType;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argument;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.argumentCount;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.call;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.functionName;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.type;
+import static io.trino.plugin.base.expression.ConnectorExpressionPatterns.variable;
+import static io.trino.plugin.jdbc.CaseSensitivity.CASE_SENSITIVE;
+import static io.trino.spi.type.BooleanType.BOOLEAN;
+
+public class RewriteCaseSensitiveComparison
+        implements ConnectorExpressionRule<Call, ParameterizedExpression>
+{
+    private static final Capture<Variable> LEFT = newCapture();
+    private static final Capture<Variable> RIGHT = newCapture();
+
+    private final Pattern<Call> pattern;
+
+    public RewriteCaseSensitiveComparison(Set<ComparisonOperator> enabledOperators)
+    {
+        Set<FunctionName> functionNames = enabledOperators.stream()
+                .map(ComparisonOperator::getFunctionName)
+                .collect(toImmutableSet());
+
+        pattern = call()
+                .with(type().equalTo(BOOLEAN))
+                .with(functionName().matching(functionNames::contains))
+                .with(argumentCount().equalTo(2))
+                .with(argument(0).matching(variable().with(type().matching(VarcharType.class::isInstance)).capturedAs(LEFT)))
+                .with(argument(1).matching(variable().with(type().matching(VarcharType.class::isInstance)).capturedAs(RIGHT)));
+    }
+
+    @Override
+    public Pattern<Call> getPattern()
+    {
+        return pattern;
+    }
+
+    @Override
+    public Optional<ParameterizedExpression> rewrite(Call expression, Captures captures, RewriteContext<ParameterizedExpression> context)
+    {
+        ComparisonOperator comparison = ComparisonOperator.forFunctionName(expression.getFunctionName());
+        Variable firstArgument = captures.get(LEFT);
+        Variable secondArgument = captures.get(RIGHT);
+
+        if (!isCaseSensitive(firstArgument, context) || !isCaseSensitive(secondArgument, context)) {
```

Comment:
nit: since it's always reverted, maybe `isCaseInsensitive(first) || isCaseInsensitive(second)`

Reply:
that was my previous version. 
but then, the method name has the inversion, so it's harder to parse  (intellij inspection are happy, but humans don't get the benefit)